### PR TITLE
Do not create posts in data providers

### PR DESCRIPTION
**Pull request by Screenfeed**
Fixes https://github.com/polylang/polylang-pro/issues/1832.

This PR removes the creation of posts from `PLL_Preload_Paths_TestCase::preload_paths_provider()`.

### CODE REVIEW

File: tests/phpunit/tests/test-preload-paths.php [hunk 1]
```diff
@@ -41,26 +42,26 @@ public function test_preload_paths_with_post_editor_context( $path, $is_filtered
...
-	 * @param string       $language        The post's language slug.
-	 * @param bool         $is_translatable Whether or not the post type is translatable.
+	 * @param string|string[] $path         The preload path under test. Could be an array if provided along a HTTP method.
+	 * @param bool            $is_filtered  Whether the path should be filtered or not.
+	 * @param string          $context_post Says what type of post must be created for the context. Possible values are `translatable`, `translatable_cpt`, `untranslatable`.
+	 * @param string          $language     The post's language slug.
 	 */
-	public function test_preload_paths_in_site_editor_context( $path, $is_filtered, $post, $language, $is_translatable ) {
+	public function test_preload_paths_in_site_editor_context( $path, bool $is_filtered, string $context_post, string $language ) {
+		list( $path ) = $this->make_data_concrete( $path, $is_filtered, $context_post );
```
Comment (Hug0-Drelon) [inline tests/phpunit/tests/test-preload-paths.php:51]: I think `list()` must be avoided even in test.
Am I right @Chouby?
See also line 13.

File: tests/phpunit/includes/testcase-preload-paths.php [hunk 2]
```diff
@@ -163,15 +160,48 @@ protected function get_paths_dataset( $post ) {
...
 		);
 	}
+
+	/**
+	 * Makes the data returned by preload_paths_provider concrete by creating a post and injecting its type and ID into the given path.
+	 *
+	 * @param string|string[] $path         The preload path under test. Could be an array if provided along a HTTP method.
+	 * @param bool            $is_filtered  Whether the path should be filtered or not.
+	 * @param string          $context_post Says what type of post must be created for the context. Possible values are `translatable`, `translatable_cpt`, `untranslatable`.
+	 * @return array                        The filtered path at key 0, the post at key 1, "is the post type translatable" at key 2.
```
Comment (Hug0-Drelon) [inline tests/phpunit/includes/testcase-preload-paths.php:181]: Maybe I didn't read correctly, but I didn't see where you use keys 1 and 2. Given that maybe we could simplify?

**Review by Hug0-Drelon — CHANGES_REQUESTED** [review]

File: tests/phpunit/tests/test-preload-paths.php (hunk 1 above)
Comment (Screenfeed) [reply tests/phpunit/tests/test-preload-paths.php:51]: I think you're confusing `list()` with `extract()`: #1333.

File: tests/phpunit/includes/testcase-preload-paths.php (hunk 2 above)
Comment (Screenfeed) [reply tests/phpunit/includes/testcase-preload-paths.php:181]: They are used in `Preload_Paths_Test::test_preload_paths_with_post_editor_context()`.

File: tests/phpunit/tests/test-preload-paths.php (hunk 1 above)
Comment (Hug0-Drelon) [reply tests/phpunit/tests/test-preload-paths.php:51]: Sorry to insist, `list()` has been removed from both our code and tests I think. I didn't find any occurence of this function.

Comment (Screenfeed) [reply tests/phpunit/tests/test-preload-paths.php:51]: 253 matches in WP's core currently.

Comment (Hug0-Drelon) [reply tests/phpunit/tests/test-preload-paths.php:51]: The core isn't always well written IMHO 😅

**Review by Hug0-Drelon — APPROVED** [review]

**Pull request merged**